### ISSUE TITLE
IndexOutOfRangeException in SparseJaggedArray

### ISSUE THREAD

**Issue by WeslomPo** [genaray/Arch#207]
I using ArchEcs 1.2.7
When I added one big feature with many components to other big feature code will breaks in random places where I add new component to my entity. It start to throw an IndexOutOfRangeException in SparseJaggedArray in Add method. Index is larger than 80.
I was trying to reproduce it in Test assembly in Arch project. I added many components to one entity and it will break in similiar manner.

To reproduce, I lower jagged array size from 80 (64+16) to 32 (16+16):
![image](https://github.com/genaray/Arch/assets/310008/e085e418-45d6-4576-87b0-bd0d362386d9)
This is not neccesarily, but creating 80 components or 80 variants of archetypes is painful imho.

Prepared dummy components:
```
public struct Type0 { }
public struct Type1 { }
public struct Type2 { }
public struct Type3 { }
public struct Type4 { }
public struct Type5 { }
public struct Type6 { }
public struct Type7 { }
public struct Type8 { }
public struct Type9 { }
public struct Type10 { }
public struct Type11 { }
public struct Type12 { }
public struct Type13 { }
public struct Type14 { }
public struct Type15 { }
public struct Type16 { }
public struct Type17 { }
public struct Type18 { }
public struct Type19 { }
public struct Type20 { }
public struct Type21 { }
public struct Type22 { }
public struct Type23 { }
public struct Type24 { }
public struct Type25 { }
public struct Type26 { }
public struct Type27 { }
public struct Type28 { }
public struct Type29 { }
```
Test:
```
[Test]
    public void SomeTest3()
    {
        var a = _world.Create<Type0>();
        _world.Add<Ai>(a);
        _world.Add<Rotation>(a);
        _world.Add<Type0>(a);
        _world.Add<Type1>(a);
        _world.Add<Type2>(a);
        _world.Add<Type3>(a);
        _world.Add<Type4>(a);
        _world.Add<Type5>(a);
        _world.Add<Type6>(a);
        _world.Add<Type7>(a);
        _world.Add<Type8>(a);
        _world.Add<Type9>(a);
        _world.Add<Type10>(a);
        _world.Add<Type11>(a);
        _world.Add<Type12>(a);
        _world.Add<Type13>(a);
        _world.Add<Type14>(a);
        _world.Add<Type15>(a);
        _world.Add<Type16>(a);
        _world.Add<Type17>(a);
        _world.Add<Type18>(a);
        _world.Add<Type19>(a);
        _world.Add<Type20>(a);
        _world.Add<Type21>(a);
        _world.Add<Type22>(a);
        _world.Add<Type23>(a);
        _world.Add<Type24>(a);
        _world.Add<Type25>(a);
        _world.Add<Type26>(a);
        _world.Add<Type27>(a);
        _world.Add<Type28>(a);
        _world.Add<Type29>(a);
    }
```
And then it will throw similiar exception.

As temporary solution I increased jagged array size from 64 to 128 elements...
Also, I was trying to optimize my archetype count by concatenate several places where components are added in next line or it is known that entity will have that component - it also helps, but not so much.

**Comment by genaray** [comment]
Have you actually tested the pre-release alpha version? That might solve the problem :) Otherwise I'll have a look at it.

**Comment by genaray** [comment]
> No.. I will look into today and will post a feedback, if it solve issue.

And? ^^

**Comment by genaray** [comment]
Closing this one since there was no reply.

**Comment by WeslomPo** [comment]
Checked today, as I have another bug, that break build on android. So, this bug is deffinitelly is solved in 1.2.8.
